ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_bgp.py

```diff
@@ -428,7 +465,29 @@ def _router_bgp_vlan_aware_bundles(self) -> list | None:
 
         bundles = []
         for tenant in self.shared_utils.filtered_tenants:
+            l2vlan_svi_vlan_aware_bundles = {}
             for vrf in tenant["vrfs"]:
+                sorted_vlan_list = sorted(vrf["svis"], key=self._get_vlan_aware_bundle_name_tuple)
+                bundle_groups = itertools_groupby(sorted_vlan_list, self._get_vlan_aware_bundle_name_tuple)
+
+                # Find SVIs which have an evpn_vlan_bundle defined
+                for vlan_aware_bundle_name_tuple, svis in bundle_groups:
+                    bundle_name, is_evpn_vlan_bundle = vlan_aware_bundle_name_tuple
+                    svis = list(svis)
+                    if is_evpn_vlan_bundle:
+                        # check if the referred name exists in the global evpn_vlan_bundles
+                        evpn_vlan_bundle = self._get_evpn_vlan_bundle(svis[0], bundle_name)
+
+                        if bundle_name in l2vlan_svi_vlan_aware_bundles:
+                            l2vlan_svi_vlan_aware_bundles[bundle_name]["l2vlan_svis"].extend(svis)
+                        else:
+                            l2vlan_svi_vlan_aware_bundles[bundle_name] = {"evpn_vlan_bundle": evpn_vlan_bundle, "l2vlan_svis": svis}
+
+                        # Remove the SVIs which have an evpn_vlan_bundle from the VRF list
+                        for bundle_svi in svis:
+                            vrf["svis"].remove(bundle_svi)
```

Comment:
Why are you removing SVIs here? This will modify the original data, and should be avoided. It would be better to give a default value to the 
`_get_vlan_aware_bundle_name_tuple()` function, so you can give it the vrf-specific bundle. This way we don't have to duplicate the logic and parse through the lists multiple times.